### PR TITLE
New optional parameter (selectPaths) for query specs

### PR DESCRIPTION
**Pull request by wmathurin**
If you specify [path1, path2] you get back [soupElt[path1], soupElt[path2]] for each matching row instead of the soupElt's.

### CODE REVIEW

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/QuerySpec.java [hunk 1]
```diff
@@ -81,8 +88,9 @@
     public final String likeKey;
 
     // Private constructor for soup query spec
-    private QuerySpec(String soupName, String path, QueryType queryType, String matchKey, String beginKey, String endKey, String likeKey, String orderPath, Order order, int pageSize) {
+    private QuerySpec(String soupName, String[] selectPaths, QueryType queryType, String matchKey, String beginKey, String endKey, String likeKey, String orderPath, Order order, int pageSize, String path) {
```
Comment (wmathurin) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/QuerySpec.java]: Keeping old APIs

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/QuerySpec.java [hunk 2]
```diff
@@ -243,20 +326,18 @@ private String computeIdsSmartSql(String smartSql) {
      * @return select clause for exact/like/range/match queries
      */
     private String computeSelectClause() {
-        return SELECT + computeFieldReference(SmartSqlHelper.SOUP) + " ";
+        List<String> fieldReferences = new ArrayList<>();
```
Comment (wmathurin) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/QuerySpec.java]: That's where it's happening

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/QuerySpec.java [hunk 3]
```diff
@@ -299,9 +380,10 @@ private String computeWhereClause() {
                     break;
                 }
             case match:
-                String joinClause = computeSoupFtsReference() + "." + SmartStore.DOCID_COL + " = " + computeFieldReference(SmartStore.SOUP_ENTRY_ID);
-                String matchCause = field + " MATCH '" + matchKey + "' "; // statement arg binding doesn't seem to work so inlining matchKey
-                pred = joinClause + " AND " + matchCause;
+                pred = computeFieldReference(SmartStore.SOUP_ENTRY_ID) + " IN ("
```
Comment (wmathurin) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/QuerySpec.java]: See https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1152/commits/99e04f5a73fe7619eaa8f3ff16208b418bfe122b for more information

**Comment by bhariharan** [comment]
👍

**Pull request merged**